src/pages/ProductPage/ProductPage.tsx

```diff
@@ -1,44 +1,93 @@
 import Button from "@mui/material/Button";
 import CartCounter from "../../components/CartCounter/CartCounter";
 import { useLoaderData } from "react-router";
 import { getProduct } from "../service";
+import { useCart } from "../CartPage/CartContext";
 
 import styles from "./ProductPage.module.scss";
 
 import { Product } from "../types";
-import { LoaderFunction } from "react-router-dom";
 
 export const loader: LoaderFunction = async ({ params }) => {
   if (!params.productId) throw new Error();
   return await Promise.resolve(getProduct(params.productId as string));
 };
 
 const ProductPage = () => {
-  const product = useLoaderData() as Product;
+  const { id, image, title, price, description } = useLoaderData() as Product;
+
+  const { cart, addItem, increaseAmount, decreaseAmount } = useCart();
+  const [initialAmount, setInitialAmount] = useState(1);
+  const isProductInCart = cart.some((product) => product.id === id);
+  const productInCart = cart.find((product) => product.id === id);
+
+  let amount: number | undefined = isProductInCart
+    ? productInCart?.amount
+    : initialAmount;
+
+  const addToCart = () => {
+    addItem({
+      id,
+      price,
+      title,
+      image,
+      amount: amount || initialAmount,
+    });
+  };
+
+  const incAmountPayload = () => {
+    if (!amount) return;
+    increaseAmount({
+      id,
+      amount: isProductInCart ? amount : initialAmount,
+    });
+  };
+
+  const decAmountPayload = () => {
+    if (!amount) return;
+    decreaseAmount({
+      id,
+      amount: isProductInCart ? amount : initialAmount,
+    });
+  };
+  const incCounter = () => {
+    setInitialAmount(initialAmount + 1);
+  };
+
+  const decCounter = () => {
+    if (initialAmount > 1) setInitialAmount(initialAmount - 1);
+  };
 
   return (
     <div className={styles.product}>
       <Grid container spacing={6}>
         <Grid xs={12} sm={5} md={6}>
           <div className={styles.img}>
-            <img src={product.image} alt={product.title} />
+            <img src={image} alt={title} />
           </div>
         </Grid>
         <Grid xs={12} sm={7} md={6}>
           <h1>
-            <FavoriteBorder /> {product.title}
+            <FavoriteBorder /> {title}
           </h1>
-          <h2>$ {product.price}</h2>
-          <h3>{product.description}</h3>
+          <h2>$ {price}</h2>
+          <h3>{description}</h3>
           <>
             <Grid container spacing={2}>
               <Grid xs={4} md={3}>
-                <CartCounter />
+                <CartCounter
+                  currentValue={amount || initialAmount}
+                  onIncrease={isProductInCart ? incAmountPayload : incCounter}
+                  onDecrease={isProductInCart ? decAmountPayload : decCounter}
+                />
               </Grid>
               <Grid xs={8} md={9}>
-                <Button fullWidth variant="outlined">
+                <Button fullWidth variant="outlined" onClick={addToCart}>
```

Comment:
I suggest you to hide the button if you already have this item in the cart and work with cart state directly
```suggestion
                {!isProductInCart  && <Button fullWidth variant="outlined" onClick={addToCart}>}
```